internal/kafka/internal/services/kafkatlscertmgmt/kafka_tls_certificate_management_service.go

```diff
@@ -186,41 +200,51 @@ func NewKafkaTLSCertificateManagementService(
 
 	return &kafkaTLSCertificateManagementService{
 		storage:              storage,
-		config:               kafkaTLSCertificateManagementConfig,
 		certManagementClient: certManagementClient,
+		config:               kafkaTLSCertificateManagementConfig,
 	}, err
 }
 
 func createCertMagicClient(awsConfig *config.AWSConfig,
 	kafkaTLSCertificateManagementConfig *config.KafkaTLSCertificateManagementConfig,
 	storage certmagic.Storage) *certmagic.Config {
 	provider := &route53.Provider{
-		WaitForPropagation: false,
-		AccessKeyId:        awsConfig.Route53.AccessKey,
-		SecretAccessKey:    awsConfig.Route53.SecretAccessKey,
+		WaitForPropagation: true,
+		// wait up to 150 seconds for the temporary txt record to be propagated.
+		// this is a blocking operation but it is acceptable since the management of certificate for each domain is done async
+		MaxWaitDur:      150 * time.Second,
```

Comment:
We wait for the txt domain to be INSYNC for up to 150 seconds

Reply:
Do we have some insight or documented information about the approximate time range it takes to propagate?

Reply:
What happens if that doesn't happen in that specified time? is an error returned?

Reply:
> Do we have some insight or documented information about the approximate time range it takes to propagate usually?

From https://aws.amazon.com/route53/faqs/ they say 60s under normal condition.

```
Q. How quickly will changes I make to my DNS settings on Amazon Route 53 propagate globally?

Amazon Route 53 is designed to propagate updates you make to your DNS records to its world-wide network of authoritative DNS servers within 60 seconds under normal conditions. A change is successfully propagated world-wide when the API call returns an INSYNC status listing.

Note that caching DNS resolvers are outside the control of the Amazon Route 53 service and will cache your resource record sets according to their time to live (TTL). The INSYNC or PENDING status of a change refers only to the state of Route 53’s authoritative DNS servers.
```

> What happens if that doesn't happen in that specified time? is an error returned?

Yes, an error will happen.

Reply:
> Yes, an error will happen.

Just to extend on what I said here - an error will happen but this won't mean a Kafka will fail to be reconciled. The reconciler will will output that error message and on the next reconciler run, we'll attempt to reconcile the cert again up until it succeeds.